### ISSUE TITLE
Expose a function to update the meta data in the readme

### ISSUE THREAD

**Issue by lvwerra** [huggingface/huggingface_hub#835]
As discussed in https://github.com/huggingface/evaluate/issues/6 we would like a function to update the meta data in the readme of a repo:

```python
huggingface_hub.metadata_update(
  repo_id="lvwerra/my_model",
  update={"metrics": new_value}
)
```

This would enable to programmatically update meta data such as metrics from the `evaluate`, `datasets`, or `transformers` libraries.

There are two use-cases:
1. The key does not exist yet so we add an entry.
2. The key does already exist but needs to be updated - in that case we might want a `overwrite=True` option as safety. 

What do you think @julien-c, @osanseviero? I can have a look at this if you want.

**Comment by julien-c** [comment]
sounds good to me! 

this can use the existing metadata-related functions under the hood, and maybe rely on upload_file so it doesn't need a local install of `git`

**Comment by osanseviero** [comment]
Yes, I think using single file download + `metadata_load`/`metadata_save` (https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/repocard.py) + `upload_file` would be very good idea
